### PR TITLE
Fix #179 - Replace autocomplete for contacts

### LINKED ISSUE #179: Replace autocomplete for contacts
We use an autocomplete to select who to send an email to. It doesn't work great atm.

This seems to work well:
https://shadcn-tag-input.vercel.app/

### PR DESCRIPTION
**Pull request by madhav-relish**
Fix #179 

- Modal is now big and fully responsive
- Simplified the styling
- changed the design of the "To" input to the desired ComboBox.

Screenshot 2024-08-27 222454

### CODE REVIEW

File: apps/web/app/(app)/compose/ComposeEmailForm.tsx [hunk 1]
```diff
@@ -166,12 +171,12 @@ export const ComposeEmailForm = (props: {
...
                 nullable={true}
               >
-                <div className="flex w-full flex-wrap items-center">
+                <div className="flex min-h-10 w-full flex-1 flex-wrap items-center gap-2 rounded-md border border-gray-300 px-2 py-2 shadow-sm focus-within:border-black focus-within:ring-black disabled:cursor-not-allowed disabled:bg-gray-50 disabled:text-gray-500 disabled:ring-gray-200 sm:text-sm">
                   {selectedEmailAddressses.map((emailAddress) => (
                     <Badge
                       key={emailAddress}
                       variant="outline"
-                      className="mr-1.5"
+                      className="h-8 rounded-md border-primary bg-primary text-white"
```
Comment (elie222) [inline apps/web/app/(app)/compose/ComposeEmailForm.tsx:179]: avoid primary and just go with the color. we don't use bg-primary anywhere in the codebase

File: apps/web/components/Modal.tsx [hunk 2]
```diff
@@ -61,25 +61,24 @@ export function Modal(props: ModalProps) {
               leave="ease-in duration-200"
               leaveFrom="opacity-100 scale-100"
               leaveTo="opacity-0 scale-95"
+              className={clsx(
```
Comment (elie222) [inline apps/web/components/Modal.tsx:64]: modals everywhere else on the site still work fine after this change?

**Review by elie222 — CHANGES_REQUESTED** [review]
A few questions

File: apps/web/app/(app)/compose/ComposeEmailForm.tsx (hunk 1 above)
Comment (madhavsingh203) [reply apps/web/app/(app)/compose/ComposeEmailForm.tsx:179]: Sure!

File: apps/web/components/Modal.tsx (hunk 2 above)
Comment (madhavsingh203) [reply apps/web/components/Modal.tsx:64]: I've tested the modals, only VideoDemo.tsx (Landing page demo video) had a layout issue which will be fixed in the next commit!

Comment (madhav-relish) [reply apps/web/components/Modal.tsx:64]: Yeah, the only place where layout was not looking good was the VideoDemo.tsx modal (Demo video on the landing page) which is now fixed in the latest PR #201

File: apps/web/app/(app)/compose/ComposeEmailForm.tsx (hunk 1 above)
Comment (madhav-relish) [reply apps/web/app/(app)/compose/ComposeEmailForm.tsx:179]: Done!

**Pull request merged**